KitePrintSDK/src/main/java/ly/kite/KiteSDK.java

```diff
@@ -306,9 +336,15 @@ static private void setParameter( Context context, Scope scope, String prefix, S
    *****************************************************/
   static private boolean getBooleanParameter( Context context, Scope scope, String prefix, String name, boolean defaultValue )
     {
-    String key = getParameterKey( prefix, name );
+      //////// Decryption initialiser //////////
+      SecurePreferences pref = new SecurePreferences(ENCRYPTION_KEY);
+
+    String key = pref.encrypt(getParameterKey( prefix, name ));//Re-encrypt the key s.t. it will match the one stored in the preference files
 
-    return ( scope.sharedPreferences( context ).getBoolean( key, defaultValue ) );
+    if( pref.decrypt(scope.sharedPreferences( context ).getString( key, ""+defaultValue )).equals("false"))
```

Comment:
That does not work because the function is supposed to return a boolean value but "pref.decrypt(scope.sharedPreferences( context ).getString( key, ""+defaultValue )).equals("false");" returns a string , therefore the if condition was put in place.